### ISSUE TITLE
Incorrect E012 report for TripUpdates

### ISSUE THREAD

**Issue by thekaveman** [CUTR-at-USF/gtfs-realtime-validator#239]
**Summary:** 

The validator incorrectly reported an [`E012`](https://github.com/CUTR-at-USF/gtfs-realtime-validator/blob/master/RULES.md#e012) violation for a TripUpdates feed.

**Steps to reproduce:** 

I'm not sure if/how this can be reproduced. The feeds in question can be found here:

[GTFS static](http://gtfs.bigbluebus.com/current.zip) - http://gtfs.bigbluebus.com/current.zip

[GTFS-rt TripUpdates](http://gtfs.bigbluebus.com/tripupdates.bin) - http://gtfs.bigbluebus.com/tripupdates.bin

[GTFS-rt VehiclePositions](http://gtfs.bigbluebus.com/vehiclepositions.bin) - http://gtfs.bigbluebus.com/vehiclepositions.bin

**Platform:** 

OS: Windows 10.0.10586
Java: 1.8.0_131-b11
Browser: Chrome 59.0.3071.115

Here is a snippet of the feed JSON as provided by the validator (on the iteration details page), showing the first of 11 trips that were part of the error report. The [complete JSON file](https://github.com/CUTR-at-USF/gtfs-realtime-validator/files/1109954/iteration_737_E012_tripupdates.txt) is attached:

```json
{
  "header": {
    "gtfs_realtime_version": "1.0",
    "incrementality": "FULL_DATASET",
    "timestamp": 1498675084
  },
  "entity": [
   {
      "id": "4095",
      "trip_update": {
        "trip": {
          "trip_id": "701392"
        },
        "stop_time_update": [
          {
            "stop_sequence": 1,
            "arrival": {
              "delay": -180
            }
          }
        ],
        "vehicle": {
          "id": "4095"
        },
        "timestamp": 1498675079
      }
    }
  ]
}
```

And the table of `E012` reports for this iteration:

OccurrenceId|Rule ID|Summary
---------------|--------|-----
1|E012|trip_id 701392 timestamp 1498675079 is greater than the header
2|E012|trip_id 701441 timestamp 1498675079 is greater than the header
3|E012|trip_id 703476 timestamp 1498675079 is greater than the header
4|E012|trip_id 703475 timestamp 1498675079 is greater than the header
5|E012|trip_id 702878 timestamp 1498675078 is greater than the header
6|E012|trip_id 700376 timestamp 1498675078 is greater than the header
7|E012|trip_id 700394 timestamp 1498675078 is greater than the header
8|E012|trip_id 701918 timestamp 1498675078 is greater than the header
9|E012|trip_id 701845 timestamp 1498675078 is greater than the header
10|E012|trip_id 702146 timestamp 1498675078 is greater than the header
11|E012|trip_id 702194 timestamp 1498675078 is greater than the header

**Comment by barbeau** [comment]
@thekaveman Thanks for the bug report!  We'll take a look.  You were monitoring both the TripUpdates and VehiclePositions feeds within the same session, right?

Currently we're grabbing a header timestamp from one of the two feeds for use in rules like E012, and I think this is the cause of the problem.  Sounds like the one feed is lagging a bit behind the other.  We should probably be using the newest (i.e., largest number) of the two (or more) timestamps when validating for rules like E012.

**Comment by barbeau** [comment]
@thekaveman Ok, I believe this should be fixed in the master branch now.  Were you running the tool locally on your own machine or on our demo server?

**Comment by thekaveman** [comment]
@barbeau Thanks! You're correct, I was monitoring both TripUpdates and VehiclePositions in the same session, and was leaning towards a conclusion like what you found.

I was running the tool locally - is your fix propagated to the `.jar`?

**Comment by barbeau** [comment]
Yes, the JAR linked in the README at https://s3.amazonaws.com/gtfs-rt-validator/travis_builds/gtfs-rt-validator-1.0.0-SNAPSHOT.jar is automatically generated by the [Travis build server](https://travis-ci.org/CUTR-at-USF/gtfs-realtime-validator), which takes between 2-5 minutes to build following a commit to the master branch.  So the above link to the JAR should always contain the most recent version of the master branch, minus the 2-5 minutes it takes Travis to build new versions.

I'll update our demo server too, I'll just wait to do so at a time to avoid disrupting people trying it out.
